tests/tlstest.py

```diff
@@ -6,17 +6,71 @@
 import suite.test_requirements as reqs
 import infra.e2e_args
 import subprocess
+import os
+import difflib
+import re
+
+
+def compare_golden():
+    script_path = os.path.realpath(__file__)
+    script_dir = os.path.dirname(script_path)
+    golden_file = os.path.join(script_dir, "tls_report.log")
+    print("Comparing output to golden file " + golden_file)
+
+    # Read both files into arrays
+    with open(golden_file, encoding="UTF-8") as g:
+        golden = g.readlines()
+    with open("tls_report.log", encoding="UTF-8") as o:
+        output = o.readlines()
+
+    # Golden file is clean of random output (IP address, hashes)
+    # New log needs to clean of random output
+    reCOMMENT = re.compile(r"^#")
+    reIPADDR = re.compile(r"\d+\.\d+\.\d+\.\d+:?\d*")
+    reFIELDS = re.compile(r"(Serial|Fingerprints|subjectAltName|Certificate Validity)")
+    reHASH = re.compile(r"SHA\d+ [0-9A-F]{20,}")
+    filtered = list()
```

Comment:
```suggestion
    filtered = []
```
Always use literals when you can, they're shorter, faster, and can't be shadowed.